### DISCUSSION TITLE
How to convert a server component into static html?

### DISCUSSION THREAD

**Discussion by j4tmr** [vercel/next.js#55135] [Help]
### Summary

Many times we need to save static html content.
I tried some methods, such as using renderToString, etc., but failed to solve the problem. See if anyone here has a solution.

**Comment by icyJoseph** [comment]
Can you give a specific use case? Like with pseudo-code?

**Reply by j4tmr** [reply]
In some cases, the front end not only uses react or based on nextjs, but also requires the same html content for display. Of course, my scenario is mainly cms and other blog content.
very simple code
```
// Component.tsx
const Component = async () => {
  const data = await getData()
  return (
    <h1>title</h1>
    <!-- content based on data -->
  )
}
```
```
// route.ts
import Component ...
export async function GET(request: Request) {
  // get the static html of the Component, 
  return NextResponse.json({ res: "ok" });
}

```

**Reply by icyJoseph** [reply]
Try something like this:

```tsx
// api/route.tsx
import { Pokemon } from "../Pokemon";

export const runtime = "edge";

export const GET = async () => {
  const { renderToReadableStream } = await import("react-dom/server");

  const stream = await renderToReadableStream(<Pokemon />, {});

  return new Response(stream, {
    headers: { "content-type": "text/html" },
  });
};

```

**Reply by j4tmr** [reply]
I tried. I found that my previous mistakes were mainly due to the native components of nextjs such as Link and Image, which may have other logic.

**Reply by icyJoseph** [reply]
what kind of errors did you see?

**Reply by j4tmr** [reply]
Thank you for your help, I am not dealing with this for the time being, I replaced with the original a and img and temporarily solved the problem.
